database/types/settings.go

```diff
@@ -0,0 +1,218 @@
+)
+
+type (
+	// Platform is the database representation of platform settings.
+	Platform struct {
+		ID sql.NullInt64 `sql:"id"`
+		Compiler
+		Queue
+
+		RepoAllowlist     pq.StringArray `json:"repo_allowlist" sql:"repo_allowlist" gorm:"type:varchar(1000)"`
+		ScheduleAllowlist pq.StringArray `json:"schedule_allowlist" sql:"schedule_allowlist" gorm:"type:varchar(1000)"`
+
+		CreatedAt sql.NullInt64  `sql:"created_at"`
+		UpdatedAt sql.NullInt64  `sql:"updated_at"`
+		UpdatedBy sql.NullString `sql:"updated_by"`
+	}
+
+	// Compiler is the database representation of compiler settings.
+	Compiler struct {
+		CloneImage        sql.NullString `json:"clone_image" sql:"clone_image"`
+		TemplateDepth     sql.NullInt64  `json:"template_depth" sql:"template_depth"`
+		StarlarkExecLimit sql.NullInt64  `json:"starlark_exec_limit" sql:"starlark_exec_limit"`
+	}
+
+	// Queue is the database representation of queue settings.
+	Queue struct {
+		Routes pq.StringArray `json:"routes" sql:"routes" gorm:"type:varchar(1000)"`
+	}
+)
+
+// Value - Implementation of valuer for database/sql for Compiler.
+func (r Compiler) Value() (driver.Value, error) {
+	valueString, err := json.Marshal(r)
+	return string(valueString), err
+}
+
+// Scan - Implement the database/sql scanner interface for Compiler.
+func (r *Compiler) Scan(value interface{}) error {
+	switch v := value.(type) {
+	case []byte:
+		return json.Unmarshal(v, &r)
+	case string:
+		return json.Unmarshal([]byte(v), &r)
+	default:
+		return fmt.Errorf("wrong type for compiler: %T", v)
+	}
+}
+
+// Value - Implementation of valuer for database/sql for Queue.
+func (r Queue) Value() (driver.Value, error) {
+	valueString, err := json.Marshal(r)
+	return string(valueString), err
+}
+
+// Scan - Implement the database/sql scanner interface for Queue.
+func (r *Queue) Scan(value interface{}) error {
+	switch v := value.(type) {
+	case []byte:
+		return json.Unmarshal(v, &r)
+	case string:
+		return json.Unmarshal([]byte(v), &r)
+	default:
+		return fmt.Errorf("wrong type for queue: %T", v)
+	}
+}
+
+// Nullify ensures the valid flag for
+// the sql.Null types are properly set.
+//
+// When a field within the Settings type is the zero
+// value for the field, the valid flag is set to
+// false causing it to be NULL in the database.
+func (s *Platform) Nullify() *Platform {
+	if s == nil {
+		return nil
+	}
+
+	// check if the ID field should be false
+	if s.ID.Int64 == 0 {
+		s.ID.Valid = false
+	}
+
+	// check if the CloneImage field should be false
+	if len(s.CloneImage.String) == 0 {
+		s.CloneImage.Valid = false
+	}
+
+	// check if the CreatedAt field should be false
+	if s.CreatedAt.Int64 < 0 {
+		s.CreatedAt.Valid = false
+	}
+
+	// check if the UpdatedAt field should be false
+	if s.UpdatedAt.Int64 < 0 {
+		s.UpdatedAt.Valid = false
+	}
+
+	return s
+}
+
+// ToAPI converts the Settings type
+// to an API Settings type.
+func (s *Platform) ToAPI() *settings.Platform {
+	ss := new(settings.Platform)
```

Comment:
super minor, but i expected the variable to be `sp` (if we're using shortform)

Reply:
how about `ps` as shorthand for the moniker "platform settings"

Reply:
went with `ps` for PlatformSettings to match the naming with `qs` and `cs` which denote QueueSettings and CompilerSettings.